### PR TITLE
remove the Spree::Api::ConfigController

### PR DESCRIPTION
**Pull request by BenMorganIO**
The `Spree::Api::ConfigController` appears to be a dummy controller. It doesn't actually have any actions. The spec (at least to me) are mysteriously passing, and its also singularly named with shorthand than expressively named in plural; `Spree::Api::ConfigController` vs `Spree::Api::ConfigurationsController`.

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://benmorganio-deprecate-config-controller.spree-spree-9ed724b77adec04c49c2.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://benmorganio-deprecate-config-controller.spree-spree-9ed724b77adec04c49c2.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by BenMorganIO** [comment]
If you use this controller, please speak up and let Spree know.

File: api/CHANGELOG.md [hunk 1]
```diff
@@ -1 +1,3 @@
-## Spree 2.4.0 (unreleased) ##
+## Spree 3.0.0 (unreleased) ##
```
Comment (JDutil) [inline api/CHANGELOG.md]: I think we should remove the changelog files, and just keep track of changes in the guides release notes of each version.

**Comment by JDutil** [comment]
:+1:

**Comment by gmacdougall** [comment]
:+1:

**Pull request closed without merging**